packages/eas-cli/src/commands/branch/publish.ts

```diff
@@ -332,32 +365,74 @@ export default class BranchPublish extends Command {
       }));
     }
 
-    let newUpdateGroup;
-    const publishSpinner = ora('Publishing...').start();
-    try {
-      newUpdateGroup = await PublishMutation.publishUpdateGroupAsync({
+    const runtimeToPlatformMapping: { [keyof: string]: string[] } = {};
+    for (const runtime of new Set(Object.values(runtimeVersions))) {
+      runtimeToPlatformMapping[runtime] = Object.entries(runtimeVersions)
+        .filter(pair => pair[1] === runtime)
+        .map(pair => pair[0]);
+    }
+
+    // Sort the updates into different groups based on their platform specific runtime versions
+    const updateGroups = Object.entries(runtimeToPlatformMapping).map(([runtime, platforms]) => {
+      const localUpdateInfoGroup = Object.fromEntries(
+        platforms.map(platform => [
+          platform,
+          unsortedUpdateInfoGroups[platform as keyof UpdateInfoGroup],
+        ])
+      );
+
+      if (republish && !oldRuntimeVersion) {
+        throw new Error(
+          'Can not find the runtime version of the update group that is being republished.'
+        );
+      }
+      return {
         branchId,
-        updateInfoGroup,
-        runtimeVersion: republish ? oldRuntimeVersion! : runtimeVersion,
+        updateInfoGroup: localUpdateInfoGroup,
+        runtimeVersion: republish ? oldRuntimeVersion : runtime,
         message,
-      });
+      };
+    });
+    let newUpdates;
+    const publishSpinner = ora('Publishing...').start();
+    try {
+      newUpdates = await PublishMutation.publishUpdateGroupAsync(updateGroups);
       publishSpinner.succeed('Published!');
     } catch (e) {
       publishSpinner.fail('Failed to published updates');
       throw e;
     }
 
     if (jsonFlag) {
-      Log.log(JSON.stringify(newUpdateGroup));
+      Log.log(JSON.stringify(newUpdates));
```

Comment:
unrelated, I added some utils to better support json flags https://github.com/expo/eas-cli/commit/094c04c7f07ba1f1ee799fafb1acbdeb4e22599a

enableJsonOutput - redirects everything to stderr
printJsonOnlyOutput - prints to real stdout

this allows someone to pipe just JSON to stdout without parsing any other non-json input
it does not work with prompts

Reply:
Nice, this seems generally useful.